Unit tests for DSP class

mvsoder opened the pull request:
Some unit tests for the DSP class to support refactor work

mvsoder committed:
Add unit tests for `DSP` class to validate processing methods

Included tests for initial state, light processing, full processing, and reset behavior. Ensures correctness of peak, RMS, spectrum normalization, and reset functionality.

mvsoder committed:
Add unit tests for `DSP` class to validate processing methods

Included tests for initial state, light processing, full processing, and reset behavior. Ensures correctness of peak, RMS, spectrum normalization, and reset functionality.

Dakkra commented on src/test/java/com/dakkra/hypersynesthesia/ffmpeg/DSPTest.java:
```diff
@@ -0,0 +1,87 @@
+        dsp.processLight(samples);
+
+        assertEquals(4, dsp.getPeak(), "Peak loudness should be absolute max of samples");
+        double expectedRms = Math.sqrt((9 + 16 + 4) / 4.0);
```
Dropping a quick comment here to explain that this is actually `sqrt((0 +9 + 16 + 4)/4.0)`  as it's supposed to be the square root of the sum of the squared values over n where n is the number of values

in other words, `n` is the array/sample buffer length

Just want the intent of the test known to future maintainers (who may not know what RMS is)

Dakkra commented on src/test/java/com/dakkra/hypersynesthesia/ffmpeg/DSPTest.java:
```diff
@@ -0,0 +1,87 @@
+        double[] spectrum = dsp.getSpectrum();
+        assertNotNull(spectrum, "Spectrum should not be null after processFull()");
+        // Expect half-size minus the 4 skipped buckets
+        assertEquals(n / 2 - 4, spectrum.length);
```
Maybe we should have the bucket skip count as a public static final field so it can be shared - in case it needs to be tweaked

mvsoder replied:
This is in a unit test, I'm not too worried about that.

Dakkra commented on src/test/java/com/dakkra/hypersynesthesia/ffmpeg/DSPTest.java:
```diff
@@ -0,0 +1,87 @@
+
+        // All values should be in [0, 1] and, for an impulse, equal after normalization
+        double max = 0;
+        for (double v : spectrum) {
```
single char variable names - didn't you teach me not to do that? ;)

mvsoder replied:
AI generated slop

Dakkra commented on src/test/java/com/dakkra/hypersynesthesia/ffmpeg/DSPTest.java:
```diff
@@ -0,0 +1,87 @@
+
+        // For impulse input, all remaining bins should normalize to 1
+        for (double v : spectrum) {
+            assertEquals(1.0, v, 1e-12);
```
I see `1e-12` used for forvigeness in floating point math in these tests, maybe make it static final as well and reference to avoid magic.

Curious what to do about denorm in the future... hmm

mvsoder replied:
Agreed

mvsoder committed:
Refactor `DSPTest` formatting and enhance consistency in style

Standardized test case formatting, added meaningful test constants, and improved readability of assertions in unit tests.
